Add basic animation to new footer and buttons

frenzibyte opened the pull request:
- Split from #28146 
- Part of #22204
- [x] Depends on #28183

Preview:

https://github.com/ppy/osu/assets/22781491/8f083979-f17d-4d30-bab2-5e1781d2bcd7

There are no animations defined anywhere in the reference designs so I rolled up my own. The intention of this PR is to at least have something there that can be tweaked.

frenzibyte committed:
Add `TopLevelContent` layer for applying external transforms on footer buttons

frenzibyte committed:
Refactor `ScreenFooter` to prepare for global usage and add transitions

frenzibyte committed:
Clean up screen footer test scene

peppy committed:
Merge branch 'master' into footer-v2-transitions

peppy reviewed (changes requested):
Looks good visually, some code concerns.

frenzibyte committed:
Move transitions inside `ScreenFooterButton` and re-use `Content` from base implementation instead

The point is to apply the transitions against a container that's inside of `ScreenFooterButton`, because the `ScreenFooterButton` drawable's position is being controlled by the flow container it's contained within, and we cannot apply the transitions on it directly.

frenzibyte commented on osu.Game/Screens/Footer/ScreenFooterButton.cs:
```diff
@@ -69,94 +69,109 @@ protected LocalisableString Text
             Size = new Vector2(BUTTON_WIDTH, BUTTON_HEIGHT);
 
-            Child = new Container
+            Child = TopLevelContent = new Container
```
I moved the animations inside of the button class as mentioned, and I've replaced `TopLevelContent` with just `Content` (which points to a local container since the button class inherits from `ClickableContainer`).

I'm not entirely sure if commenting is necessary though, the button class doesn't apply transitions on `this` anywhere else, and there's a precedent for this in `ShearedButton`, where `Content` is used to rescale the button instead of `this`, so that if there's a button next to this one all inside of a flow container, it won't move from its place.
